### PR TITLE
Dropdown Component Updates

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description

Default dropdown uses the custom dropdown UI (adding searching ability)
`max_choices` param, can be set when `multiselect` is `True`
Adding A11y accessibility 
Some styling updates (hover over tokens)

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: #3184, #3163

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: ui/packages/form/src/Dropdown.svelte [hunk 1]
```diff
@@ -1,67 +1,356 @@
 <script lang="ts">
```
Comment (pngwn) [inline ui/packages/form/src/Dropdown.svelte:1]: This file is getting too large. It needs to be broken down into smaller components, so it is easier to reason about.

**Review by pngwn — CHANGES_REQUESTED** [review]
The design of this dropdown isn't consistent with the previous one. The font size is too large when typing into the box and the padding around the text is not consistent with the other form inputs. As a result the heights of this component don't match with the heights of the other 'input' components.

Compare the dropdown to the `Number` component at the top.

Screenshot 2023-02-21 at 12 08 42

**Comment by dawoodkhan82** [comment]
@pngwn I can update the text size and padding but didn't want to interfere with the changes being made here (https://github.com/gradio-app/gradio/issues/3163). But I guess I can make those changes myself in this PR

**Comment by pngwn** [comment]
@dawoodkhan82 We can find another issue for that person. They are still having difficulties running the project locally.

**Comment by pngwn** [comment]
@Neilblaze I'll be looking into your issue this week too. We don't want to discourage contributions because things are difficult to setup.

**Comment by dawoodkhan82** [comment]
@Neilblaze Will do!

**Comment by dawoodkhan82** [comment]
"flaky" backend tests are passing locally. Not sure what's going on

**Comment by freddyaboulton** [comment]
I would ignore that failure @dawoodkhan82 !

**Review by pngwn — APPROVED** [review]
lgtm!

**Comment by dawoodkhan82** [comment]
@abidlabs mind taking a quick look?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # Upcoming Release
 
 ## New Features:
+
+### Dropdown Component Updates
+
+The standard dropdown component now supports searching for choices. Also when `multiselect` is `True`, you can specify `max_choices` to set the maximum number of choices you want the user to be able to select from the dropdown component.
+
+```python
+gr.Dropdown(label="Multiselect Dropdown", choices=CHOICES, multiselect=True, max_choices=2),
```
Comment (abidlabs) [inline CHANGELOG.md:10]: I would suggest just making this more of a complete example:

```suggestion
gr.Dropdown(label="Choose your favorite colors", choices=["red", "blue", "green", "yellow", "orange"], multiselect=True, max_choices=2),
```

File: gradio/components.py [hunk 3]
```diff
@@ -1238,6 +1240,9 @@ def __init__(
         if multiselect:
             if isinstance(value, str):
                 value = [value]
+        if not multiselect and max_choices is not None:
+            warnings.warn("Max choices is ignored when multiselect is False.")
```
Comment (abidlabs) [inline gradio/components.py:1244]:
```suggestion
            warnings.warn("The `max_choices` parameter is ignored when `multiselect` is False.")
```

**Comment by abidlabs** [comment]
Some documentation nits, but otherwise looks awesome @dawoodkhan82!

**Review by abidlabs — APPROVED** [review]

**Pull request merged**